mlir/lib/Gradient/Transforms/gradient_lowering.cpp

```diff
@@ -78,6 +100,21 @@ struct GradientLoweringPass : public OperationPass<ModuleOp> {
         if (failed(applyPatternsAndFoldGreedily(op, std::move(gradientPatterns)))) {
             return signalPassFailure();
         }
+
+        if (lowerVector) {
+            RewritePatternSet gradientVectorPatterns(&getContext());
+            ConversionTarget target(getContext());
+            target.addLegalDialect<arith::ArithDialect, memref::MemRefDialect, func::FuncDialect,
+                                   scf::SCFDialect>();
+            target.addLegalOp<UnrealizedConversionCastOp>();
+            target.addIllegalOp<VectorInitOp, VectorPushOp, VectorSizeOp, VectorLoadDataOp>();
+
+            populateVectorLoweringPatterns(vectorTypeConverter, gradientVectorPatterns);
+
+            if (failed(applyPartialConversion(op, target, std::move(gradientVectorPatterns)))) {
+                return signalPassFailure();
+            }
+        }
```

Comment:
Somewhat related to my first comment, but I think this would be well suited as a standalone pass.